src/terms/Hamiltonian.jl

```diff
@@ -117,6 +117,10 @@ end
                                                                            H::DftHamiltonianBlock,
                                                                            ψ::AbstractArray)
     n_bands = size(ψ, 2)
+    # return empty array if ψ is empty
+    if iszero(n_bands)
+        return zeros(eltype(ψ), size(ψ,1), 0)
+    end
```

Comment:
I would not bother allocating a new thing:
```suggestion
    iszero(n_bands) && return Hψ  # Nothing to do if ψ empty
```